lib/excon/connection.rb

```diff
@@ -383,19 +386,42 @@ def detect_content_length(body)
       end
     end
 
-    def validate_params(validation, params)
+    def valid_middleware_keys(middlewares)
+      middlewares.flat_map do |middleware|
+        if middleware.respond_to?(:valid_parameter_keys)
```

Comment:
Since this method is in the base class, shouldn't it always respond to the method or return `[]` already? Does that make sense? Am I missing something?

Reply:
This handles backwards compatibility if someone has already written middleware that doesn't inherit from the base class. Maybe that's not worth it and it's fair to assume that inheriting from `Base` is the only supported way to write middleware?

Reply:
That makes sense, from the beginning they were written to expect to inherit, so I think we should be ok not worrying about it. I guess I'd rather leave it out for now, and add it if we see problems (if nothing else, it would be a lot harder to remove later, and I am honestly quite likely to forget why it is here in time). Does that make sense?